reference/transactions: update pessimistic doc (#1589)
pingcap/docs: dev/reference/transactions/transaction-pessimistic.md

- TiDB supports both the optimistic transaction mode and pessimistic transaction mode in the same cluster. You can specify either mode for transaction execution.

## Usage of pessimistic transaction mode

> To enable the pessimistic transaction mode, choose any of the following three methods that suits your needs:

Reviewer: `enable` -> `apply`

- Use the `BEGIN PESSIMISTIC;` statement to start the transaction in the pessimistic transaction mode. You can write it in comment style as `BEGIN /*!90000 PESSIMISTIC */;` to make it compatible with the MySQL syntax.

- Execute the `set @@tidb_txn_mode = 'pessimistic';` statement to allow all the explicit transactions (namely non-autocommit transactions) processed in this session to be in the pessimistic transaction mode.

> - Execute the `set @@global.tidb_txn_mode = 'pessimistic';` statement to allow all newly created sessions of the entire cluster to adopt the pessimistic transaction mode and execute the explicit transactions.

Reviewer 2: ```suggestion
- Execute the `set @@global.tidb_txn_mode = 'pessimistic';` statement to allow all newly created sessions of the entire cluster to adopt the pessimistic transaction mode and execute explicit transactions.
```

Reviewer: Should be `to` or `when` instead of `and` in `and execute the explicit transactions`

Reviewer: `apply` is better than `adopt`.

> After you set `global.tidb_txn_mode` to `pessimistic`, the pessimistic transaction mode is enabled by default; but you can use the following two methods to adopt the optimistic transaction mode for the transaction:

Reviewer 2: ```suggestion
After you set `global.tidb_txn_mode` to `pessimistic`, the pessimistic transaction mode is adopted by default; but you can use either of the following two methods to adopt the optimistic transaction mode for the transaction:
```

- Use the `BEGIN OPTIMISTIC;` statement to start the transaction in the optimistic transaction mode. You can write it in comment style as `BEGIN /*!90000 OPTIMISTIC */;` to make it compatible with the MySQL syntax.

- Execute the `set @@tidb_txn_mode = 'optimistic';` statement to allow all the transactions processed in this session to be in the optimistic transaction mode.

> The `BEGIN PESSIMISTIC;` and `BEGIN OPTIMISTIC;` statements take precedence over the `tidb_txn_mode` system variable. Transactions that are opened with these two statements ignore system variables.

Reviewer 2: ```suggestion
The `BEGIN PESSIMISTIC;` and `BEGIN OPTIMISTIC;` statements take precedence over the `tidb_txn_mode` system variable. Transactions that are opened with these two statements will ignore system variables.
```

Reviewer 2: Here I suggest we add a "will" to denote the future behavior based on judgment.

> To disable the pessimistic transaction mode, you can modify the configuration file and add `enable = false` to the `[pessimistic-txn]` category.

Reviewer 2: ```suggestion
To disable the pessimistic transaction mode, modify the configuration file and add `enable = false` to the `[pessimistic-txn]` category.
```
